pkg/mlsstore/store.go

```diff
@@ -43,83 +39,57 @@ func New(ctx context.Context, config Config) (*Store, error) {
 }
 
 // Creates the installation and last resort key package
-func (s *Store) CreateInstallation(ctx context.Context, installationId []byte, walletAddress string, lastResortKeyPackage []byte, credentialIdentity []byte) error {
+func (s *Store) CreateInstallation(ctx context.Context, installationId []byte, walletAddress string, credentialIdentity, keyPackage []byte, expiration uint64) error {
 	createdAt := nowNs()
 
 	installation := Installation{
 		ID:                 installationId,
 		WalletAddress:      walletAddress,
 		CreatedAt:          createdAt,
+		UpdatedAt:          createdAt,
 		CredentialIdentity: credentialIdentity,
-	}
-
-	keyPackage := NewKeyPackage(installationId, lastResortKeyPackage, true)
-
-	return s.db.RunInTx(ctx, &sql.TxOptions{}, func(ctx context.Context, tx bun.Tx) error {
-		_, err := tx.NewInsert().
-			Model(&installation).
-			Ignore().
-			Exec(ctx)
-
-		if err != nil {
-			return err
-		}
-
-		_, err = tx.NewInsert().
-			Model(keyPackage).
-			Ignore().
-			Exec(ctx)
-
-		if err != nil {
-			return err
-		}
 
-		return nil
-	})
-}
+		KeyPackage: keyPackage,
+		Expiration: expiration,
+	}
 
-// Insert a batch of key packages, ignoring any that may already exist
-func (s *Store) InsertKeyPackages(ctx context.Context, keyPackages []*KeyPackage) error {
-	_, err := s.db.NewInsert().Model(&keyPackages).Ignore().Exec(ctx)
+	_, err := s.db.NewInsert().
+		Model(&installation).
+		Ignore().
+		Exec(ctx)
 	return err
 }
 
-func (s *Store) ConsumeKeyPackages(ctx context.Context, installationIds [][]byte) ([]*KeyPackage, error) {
-	keyPackages := make([]*KeyPackage, 0)
-	err := s.db.RunInTx(ctx, &sql.TxOptions{}, func(ctx context.Context, tx bun.Tx) error {
-		err := tx.NewRaw(`
-			SELECT DISTINCT ON(installation_id) * FROM key_packages
-			WHERE "installation_id" IN (?)
-			AND not_consumed = TRUE
-			ORDER BY installation_id ASC, is_last_resort ASC, created_at ASC
-			`,
-			bun.In(installationIds)).
-			Scan(ctx, &keyPackages)
-
-		if err != nil {
-			return err
-		}
+// Insert a new key package, ignoring any that may already exist
+func (s *Store) UpdateKeyPackage(ctx context.Context, installationId, keyPackage []byte, expiration uint64) error {
+	installation := Installation{
+		ID:        installationId,
+		UpdatedAt: nowNs(),
 
-		if len(keyPackages) < len(installationIds) {
-			return errors.New("key packages not found")
-		}
+		KeyPackage: keyPackage,
+		Expiration: expiration,
+	}
 
-		_, err = tx.NewUpdate().
-			Table("key_packages").
-			Set("consumed_at = ?", nowNs()).
-			Set("not_consumed = FALSE").
-			Where("is_last_resort = FALSE").
-			Where("id IN (?)", bun.In(extractIds(keyPackages))).
-			Exec(ctx)
+	_, err := s.db.NewUpdate().
```

Comment:
I don't believe this will return an error if no installations matched the update query. We might want to change this to error out in that case, since the caller would want to know that their request did nothing.

Reply:
This change was documented in the proto PR: https://github.com/xmtp/proto/pull/123/files#diff-89ac5361df54d78c25ce625d56f63dd01e1bd813456a5430d992ff33c8fc3a2dR143

It seemed brittle to have the whole request fail if one installation doesn't exist. I can remove if it would be an issue though

Reply:
Nevermind, I thought comment this was on FetchKeyPackages. Let me try to fix